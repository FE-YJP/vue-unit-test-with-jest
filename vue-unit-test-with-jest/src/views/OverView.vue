<template>
  <div>
    <h1 class="title">this is overview</h1>
    <div class="example">
      <h2 class="example_title">filter的单元测试例</h2>
      <el-input class="input" v-model="userInput" placeholder="输入长度大于7的内容，下方将显示过滤后内容"></el-input>
      <div v-show="toggleShow" class="text">{{userInput | formatText}}</div>
      <div v-show="!toggleShow" class="text">{{userInput}}</div>
      <app-button @click="changeShow">切换显示方式</app-button>
    </div>
    <div class="example">
      <h2 class="example_title">路由跳转的单元测试例</h2>
      <app-button size="lg" @click="goVIPs">跳转到VIP用户页面</app-button>
    </div>
    <div class="example">
      <h2 class="example_title">Axios的单元测试例</h2>
      <app-button size="lg" @click="getData">从后台接口获取数据</app-button>
    </div>
  </div>
</template>
<script>
export default {
  data () {
    return {
      userInput: '121',
      toggleShow: true,
      usersInfo: []
    }
  },
  filters: {
    formatText: function (value) {
      console.log(value)
      if (!value) return ''
      if (value.length > 7) {
        let frontVal = value.slice(0, 2)
        let endVal = value.slice(value.length - 2, value.length)
        return frontVal + '...' + endVal
      } else {
        return value
      }
    }

  },
  methods: {
    changeShow () {
      this.toggleShow = !this.toggleShow
    },
    goVIPs () {
      this.$router.push({
        path: '/vips'
      })
    },
    getData () {
      this.$axios
        .get('users/')
        .then(res => {
          this.usersInfo = res.data
        })
        .catch(err => console.log(err))
        .finally()
    }
  }
}
</script>
<style scoped>
.title{
  margin-bottom: 20px;
}
.example{
  text-align: center;
  padding: 20px;
  border: 1px dashed red;
  margin-bottom: 50px;
}
.example_title{
  margin-bottom: 20px;
  color:blueviolet;
}
.input{
  width: 500px;
  margin-bottom: 20px;
}
.text{
  width: 500px;
  height: 30px;
  background-color:darkkhaki;
  margin: 0 auto;
  line-height: 30px;
  margin-bottom: 20px;
  font-weight: 700;
  color:deeppink;
}
</style>
