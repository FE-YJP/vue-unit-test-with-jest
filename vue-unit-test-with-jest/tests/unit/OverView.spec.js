/**
 * OverView.vue组件
 * 测试内容包括以下
 * 临时vue实例上挂载vue-router和element-ui
 * v-show
 * filter
 * axios
 * router
 * vuex
 */
import { shallowMount, createLocalVue } from '@vue/test-utils'
import OverView from '@/views/OverView.vue'
import VueRouter from 'vue-router'
import Element from 'element-ui'

// 创建临时Vue实例，挂载组件中使用的插件
const localVue = createLocalVue()
localVue.use(VueRouter)
localVue.use(Element)

describe('OverView.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(OverView, { localVue,
      stubs: ['app-button']
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  // 测试内容：snapshot->概括的测试DOM结构
  // 如果组件内存在比较特殊的需要测试的DOM结构的话，可以单独测试（详见AppButton测试文件末尾）
  it('snapshot test', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
