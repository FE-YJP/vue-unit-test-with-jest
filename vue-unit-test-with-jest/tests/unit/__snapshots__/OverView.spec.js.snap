// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverView.vue snapshot test 1`] = `
<div>
  <h1 class="title">this is overview</h1>
  <div class="example">
    <h2 class="example_title">filter的单元测试例</h2>
    <el-input-stub value="" type="text" autocomplete="off" validateevent="true" placeholder="输入长度大于7的内容，下方将显示过滤后内容" class="input"></el-input-stub>
    <div class="text"></div>
    <div class="text" style="display: none;"></div>
    <app-button-stub>切换显示方式</app-button-stub>
  </div>
  <div class="example">
    <h2 class="example_title">路由跳转的单元测试例</h2>
    <app-button-stub size="lg">跳转到VIP用户页面</app-button-stub>
  </div>
  <div class="example">
    <h2 class="example_title">Axios的单元测试例</h2>
    <app-button-stub size="lg">从后台接口获取数据</app-button-stub>
  </div>
</div>
`;
